php/class-import.php

```diff
@@ -0,0 +1,247 @@
+<?php
+/**
+ * Block Lab Importer.
+ *
+ * @package   Block_Lab
+ * @copyright Copyright(c) 2018, Block Lab
+ * @license   http://opensource.org/licenses/GPL-2.0 GNU General Public License, version 2 (GPL-2.0)
+ */
+
+namespace Block_Lab;
+
+use Block_Lab\Blocks\Block;
+
+/**
+ * Class Import
+ */
+class Import extends Component_Abstract {
+
+	/**
+	 * Importer slug.
+	 *
+	 * @var string
+	 */
+	public $slug = 'block-lab';
+
+	/**
+	 * Initialise the Upgrade component.
+	 */
+	public function init() {
+	}
+
+	/**
+	 * Register any hooks that this component needs.
+	 */
+	public function register_hooks() {
+		add_action( 'admin_init', array( $this, 'register_importer' ) );
+	}
+
+	/**
+	 * Register the importer for the Tools > Import admin screen
+	 */
+	public function register_importer() {
+		register_importer(
+			$this->slug,
+			__( 'Block Lab', 'block-lab' ),
+			__( 'Import custom blocks created with Block Lab.', 'block-lab' ),
+			array( $this, 'render_page' )
+		);
+	}
+
+	/**
+	 * Render the import page. Manages the three separate stages of the JSON import process.
+	 */
+	public function render_page() {
```

Comment:
The import UI for the pro version looks good, though I didn't test it with a JSON file:


![import-ui](https://user-images.githubusercontent.com/4063887/53199106-cde9a300-35e3-11e9-8c6f-00d779d7f34b.gif)